pkg/odo/cli/url/create.go

```diff
@@ -157,17 +177,18 @@ func (o *URLCreateOptions) Validate() (err error) {
 	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
 		// if experimental mode is enabled, and devfile is provided.
 		// check if valid host is provided
-		if len(o.host) <= 0 {
-			return fmt.Errorf("host must be provided in order to create ingress")
-		}
-		for _, localURL := range o.EnvSpecificInfo.GetURL() {
-			curIngressDomain := fmt.Sprintf("%v.%v", o.urlName, o.host)
-			ingressDomainEnv := fmt.Sprintf("%v.%v", localURL.Name, localURL.Host)
-			if curIngressDomain == ingressDomainEnv {
-				return fmt.Errorf("the url %s already exists in the application: %s", curIngressDomain, o.Application)
+		if !pushtarget.IsPushTargetDocker() {
+			if len(o.host) <= 0 {
```

Comment:
please also add a check that the exposed port flag is not used when the push target is not docker